src/lib.rs

```diff
@@ -394,15 +398,41 @@ fn internal_publish(
 
 /// List the packages from the workspace in the order of their dependencies.
 ///
-/// The list of pacakges will be written to `output`. If `manifest_path` is provided
+/// The list of packages will be written to `output`. If `manifest_path` is provided
 /// then it is expected to give the path to the `Cargo.toml` file for the root of the
 /// workspace. If `manifest_path` is `None` then `list_packages` will look for the
 /// root of the workspace in a `Cargo.toml` file in the current directory.
 ///
 /// This is a debuging aid and does not directly correspond to a sementic release
 /// step.
 pub fn list_packages(
+    #[cfg(not(feature = "napi-rs"))] output: impl Write,
+    manifest_path: Option<impl AsRef<Path>>,
+) -> Result<()> {
+    internal_list_packages(output, &PublishArgs::default(), manifest_path)
+}
+
+/// List the packages from the workspace in the order of their dependencies as
+/// matched against an argument set.
+///
+/// The list of packages will be written to `output`. If `manifest_path` is provided
+/// then it is expected to give the path to the `Cargo.toml` file for the root of the
+/// workspace. If `manifest_path` is `None` then `list_packages` will look for the
+/// root of the workspace in a `Cargo.toml` file in the current directory.
+///
+/// This is a debuging aid and does not directly correspond to a sementic release
+/// step.
+pub fn list_packages_with_arguments(
```

Comment:
I opted to add an additional public function here with identical behavior to list_packages to support the new registry functionality without introducing a breaking change.